internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1384,25 +1339,29 @@ private CompletableFuture<PubSubProduceResult> sendMessage(
   /**
    * We only include the protocol schema headers on this writer's first message to each partition.
    */
+  private PubSubMessageHeaders getHeaders(ProducerMetadata producerMetadata) {
+    return getHeaders(producerMetadata, false, LeaderCompleteState.LEADER_NOT_COMPLETED);
+  }
+
   private PubSubMessageHeaders getHeaders(
       ProducerMetadata producerMetadata,
-      boolean addLeaderCompletedStatus,
-      int partitionId) {
+      boolean addLeaderCompleteState,
+      LeaderCompleteState leaderCompleteState) {
     PubSubMessageHeaders returnPubSubMessageHeaders;
     PubSubMessageHeaders pubSubMessageHeaders =
         producerMetadata.getSegmentNumber() == 0 && producerMetadata.getMessageSequenceNumber() == 0
             ? protocolSchemaHeaders
             : EMPTY_MSG_HEADERS;
 
-    if (addLeaderCompletedStatus) {
+    if (addLeaderCompleteState) {
       // copy protocolSchemaHeaders locally and add extra header for isLeaderCompleted
       returnPubSubMessageHeaders = new PubSubMessageHeaders();
       for (PubSubMessageHeader header: pubSubMessageHeaders.toList()) {
         returnPubSubMessageHeaders.add(header);
       }
       // 1 byte holding 0/1 basd on isLeaderCompleted
       byte[] val = new byte[1];
-      val[0] = (byte) (getPartitionLeaderCompletionStatus(partitionId) ? 1 : 0);
+      val[0] = (byte) ((leaderCompleteState == LeaderCompleteState.LEADER_COMPLETED) ? 1 : 0);
```

Comment:
I think it would look cleaner if the numerical value was a property of the enum, and this code here can call `leaderCompleteState.getValue()` or the like.